lib/Frontend/ArgsToFrontendOutputsConverter.cpp

```diff
@@ -451,3 +454,77 @@ void SupplementaryOutputPathsComputer::deriveModulePathParameters(
   mainOutputIfUsable =
       canUseMainOutputForModule && !OutputFiles.empty() ? OutputFiles[0] : "";
 }
+
+static SupplementaryOutputPaths
+createFromTypeToPathMap(const TypeToPathMap *map) {
+  SupplementaryOutputPaths paths;
+  if (!map)
+    return paths;
+  static const std::vector<std::pair<types::ID, std::string &>> typesAndStrings{
```

Comment:
Please use a plain old C array for this rather than a std::vector, since it won't change sides. (It's also *definitely* not `static`, since you include references to the local variable!)

Reply:
If you really want to make it static, you can use pointers-to-members here.

Reply:
Thanks for the catch on the *static* -- d'oh! Not sure what 'change sides' means, but I'm happy to make it a C array, with some sentinel. Don't understand why a C array is better.

Reply:
Oops, that was supposed to be "size". And you don't need a sentinel; C arrays know their sizes. That means the whole thing can be a single stack allocation.

Reply:
Thanks. Fixed!